apps/newjerseypath/new_jersey_path.star

```diff
@@ -0,0 +1,238 @@
+                # path line with multiple colors.
+                # would be cool to calculate this automatically but starlark
+                # doesn't have any color or hex-printing libraries
+                render.Box(
+                    width = 1,
+                    height = 11,
+                    color = "#A6967E",
+                ),
+                render.Box(
+                    width = 5,
+                    height = 11,
+                    child = render.Padding(
+                        child = render.Circle(
+                            color = arrival["lineColors"][1],
+                            diameter = 11,
+                        ),
+                        pad = (0, 0, 6, 0),
+                    ),
+                ),
+            ],
+        )
+    else:
+        circle_widget = render.Circle(
+            color = arrival["lineColors"][0],
+            diameter = 11,
+        )
+
+    return render.Row(
+        children = [
+            render.Padding(
+                child = circle_widget,
+                pad = 2,
+            ),
+            render.Column(
+                children = [
+                    render.Marquee(
+                        child = render.Text(arrival["friendlyRouteName"]),
+                        width = 49,
+                    ),
+                    render.Text(
+                        content = wait_time_text,
+                        color = "#ffa500",
+                        offset = 1,
+                    ),
+                ],
+            ),
+        ],
+    )
+
+def get_routes(api_response):
+    routes = {}
+
+    for arrival in api_response["upcomingTrains"]:
+        route_key = "{}|{}".format(arrival["route"], arrival["direction"])
+        arrival_time = time.parse_time(arrival["projectedArrival"])
+        if route_key in routes:
+            # we've already seen this route, just stick the arrival time into it
+            routes[route_key]["arrivalTimes"].append(arrival_time)
+        else:
+            # we haven't seen this route yet, make a new entry for it
+            routes[route_key] = {
+                "friendlyRouteName": arrival["routeDisplayName"],
+                "arrivalTimes": [arrival_time],
+                "lineColors": arrival["lineColors"],
+                "direction": arrival["direction"],
+            }
+
+    routes_ordered = list(routes.values())
+
+    # sort the arrivals in chronological order
+    for route in routes_ordered:
+        route["arrivalTimes"] = sorted(route["arrivalTimes"])
+
+    # sort the routes so the one with the soonest arrival is first
+    routes_ordered = sorted(routes_ordered, key = lambda route: route["arrivalTimes"][0])
+    return routes_ordered
+
+def main(config):
+    station = config.get("station") or "grove_street"
+    desired_direction = config.get("direction") or "both"
+    path_url_for_station = PATH_URL.format(station = station)
+    api_response = http.get(path_url_for_station)
```

Comment:
We'll need some caching around this call to keep request rates down.

Reply:
Makes sense, added a per-station cache with 30s TTL